core/src/main/java/org/axonframework/commandhandling/distributed/AnnotationRoutingStrategy.java

```diff
@@ -67,7 +67,7 @@ public AnnotationRoutingStrategy() {
      *                       targeted aggregate.
      */
     public AnnotationRoutingStrategy(Class<? extends Annotation> annotationType) {
-        this(annotationType, UnresolvedRoutingKeyPolicy.ERROR);
+        this(annotationType, UnresolvedRoutingKeyPolicy.RANDOM_KEY);
```

Comment:
My guess would be to keep the previous behaviour, but allow users to set a different `UnresolvedRoutingKeyPolicy`,
Reason behind that is to not introduce unexpected behaviour for users which will move up a version.

Reply:
That was my initial reaction as well. However, random is a more sensible default, in this case.
I expect the negative effects to be minimal, but we do need to explicitly document this as a change for the next release.

An alternative would be to have the autoconfiguration auto-create an instance using the RANDOM policy and leave this to error.

Reply:
That sounds like a nice solution as the auto configuration is the only spot where we're providing a strategy.
Opting to leave this as is and adjust the auto config to provide the policy random.